blog/_posts/2019-01-18-fluxdiffeq.md

```diff
@@ -284,7 +272,7 @@ p = param([2.2, 1.0, 2.0, 0.4]) # Initial Parameter Vector
 params = Flux.Params([p])
 
 function predict_rd() # Our 1-layer neural network
-  diffeq_rd(p,reduction,prob,Tsit5(),saveat=0.1)
+  diffeq_rd(p,prob,Tsit5(),saveat=0.1)

```

Comment:
this stuff still needs to do the reduction afterwards or it won't return an array and I'm not sure the code would work.

Reply:
So `diffeq_rd(...)[1,:]`?

Reply:
I'm not totally clear if the slicing does something important here, but conversion to a vector isn't inherently necessary for the AD to work here, so it might still be possible to remove it in some cases.

Reply:
It's actually going from values in different vectors in a vector of vectors to a single contiguous vector for the time series of 1 variable. So it's not your average slice. It's not important for AD, but if someone puts the layer function in say the `Chain` code and then assumes linear algebra is going to work (putting a Dense right after), they will get an error.